Null checks outcomes android

rgomezp opened the pull request:
Motivation: Add logging in Android bridge for cases where outcome object is null to establish consistency with iOS (PR: #1129 )

rgomezp committed:
Add logging in Android bridge for cases where outcome object is null

Jeasmine commented on android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:
```diff
@@ -538,6 +538,7 @@ public void sendOutcome(final String name, final Callback callback) {
          public void onSuccess(OSOutcomeEvent outcomeEvent) {
             if (outcomeEvent == null)
                callback.invoke(new WritableNativeMap());
+               Log.e("OneSignal", "sendOutcome OSOutcomeEvent is null");
```
Maybe this can be called before `callback.invoke(new WritableNativeMap());` in that way it appears in order? Also, we might want to use OneSignal Log?

rgomezp replied:
I ended up refactoring it based on a conversation with Josh regarding it matching iOS. We are no longer invoking the callback in the null case.

For the log, we don't use the native logging method anywhere in the bridge. Any change to that should probably be a stand alone PR

jkasten2 commented on android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:
```diff
@@ -538,6 +538,7 @@ public void sendOutcome(final String name, final Callback callback) {
          public void onSuccess(OSOutcomeEvent outcomeEvent) {
             if (outcomeEvent == null)
                callback.invoke(new WritableNativeMap());
```
The null behavior differs from what is done in iOS.
https://github.com/OneSignal/react-native-onesignal/pull/1129/files#diff-153b24fed9bd79fc91c7ca30ccccdd37f29859785574f541911460e7a4cbdd6cR392

There we do NOT fire the callback at all if it is `null`. The behavior should match between Android and iOS.

rgomezp committed:
Refactor to match iOS

Motivation: we want Android bridge to match the iOS bridge. For this reason, we are removing the invocation of the callback altogether in the case that the outcome object is null.
